test/new_relic/agent/instrumentation/rails/action_controller_subscriber.rb

```diff
@@ -316,4 +322,17 @@ def test_records_span_level_error
 
     assert_segment_noticed_error txn, /controller/i, exception_class.name, /Natural 1/i
   end
+
+  def test_records_code_level_metrics
+    @subscriber.start('process_action.action_controller', :id, @entry_payload)
+    txn = NewRelic::Agent::Transaction.tl_current
+    @subscriber.finish('process_action.action_controller', :id, @entry_payload)
+    attributes = txn.segments.first.code_attributes
+
+    assert_equal __FILE__, attributes['code.filepath']
+    assert_equal 'index', attributes['code.function']
+    assert_equal 18, attributes['code.lineno']
```

Comment:
This seems like it could break easily if someone adds something above where it's currently defined (same with the tests in the other file). Is it possible for us to get the expected line number in a more flexible way?

Reply:
The easiest dynamic way to get at the correct value would be to leverage `#source_location`, just like the method being tested does. I'll put build together that does that.

Reply:
Addressed with 5cfe616 and 105f061